Sources/SpeziOpenAI/OpenAIComponent.swift

```diff
@@ -71,13 +71,13 @@ public class OpenAIComponent: Component, ObservableObject, ObservableObjectProvi
     ///   - messages: A collection of chat  messages used in the conversation.
     ///
     /// - Returns: The content of the response from the API.
-    public func queryAPI(withChat chat: [Chat]) async throws -> AsyncThrowingStream<ChatStreamResult, Error> {
+    public func queryAPI(withChat chat: [Chat], withFunction chatFunctionDeclaration: [ChatFunctionDeclaration]) async throws -> AsyncThrowingStream<ChatStreamResult, Error> {
```

Comment:
I think it might make sense to add a default value here as well, so this is not a breaking API change. We can just use a default empty Array:
```suggestion
    public func queryAPI(withChat chat: [Chat], withFunction chatFunctionDeclaration: [ChatFunctionDeclaration] = []) async throws -> AsyncThrowingStream<ChatStreamResult, Error> {
```